Carbon.IIS/Functions/Set-CIisConfigurationAttribute.ps1

```diff
@@ -74,17 +74,31 @@ function Set-CIisConfigurationAttribute
         [Microsoft.Web.Administration.ConfigurationElement] $ConfigurationElement,
 
         # A hashtable whose keys are attribute names and the values are the attribute values. Any attribute *not* in
-        # the hashtable is removed from the configuration section (i.e. reset to its default value).
+        # the hashtable is ignored, unless the `All` switch is present, in which case, any attribute *not* in the
+        # hashtable is removed from the configuration section (i.e. reset to its default value).
         [Parameter(Mandatory, ParameterSetName='AllByConfigElement')]
         [Parameter(Mandatory, ParameterSetName='AllByConfigPath')]
         [hashtable] $Attribute,
 
-        # The target element the changes is being made on. Used in messages written to the console. The default is to
+        # The target element the change is being made on. Used in messages written to the console. The default is to
         # use the type and tag name of the ConfigurationElement.
         [Parameter(ParameterSetName='AllByConfigElement')]
         [Parameter(ParameterSetName='AllByConfigPath')]
         [String] $Target,
 
+        # Properties to skip and not change. These are usually private settings that we shouldn't be mucking with or
+        # settings that capture current state, etc.
+        [Parameter(ParameterSetName='AllByConfigElement')]
+        [Parameter(ParameterSetName='AllByConfigPath')]
+        [String[]] $Exclude = @(),
+
+        # Resets *all* the target configuration element's settings to their default values, *except* each setting

```

Comment:
Perhaps include interaction with `-Exclude`.

I might expect exclude to be applied first, including to `-Reset`.

Reply:
`Reset` respects `Exclude`. The `Set-AttributeValue` function is where the Exclude parameter is handled and here's where we use `Reset` and `Set-AttributeValue`:

```powershell
        $ConfigurationElement.Attributes |
            Where-Object {
                if( $Reset )
                {
                    return $true
                }

                return $Attribute.ContainsKey($_.Name)
            } |
            Sort-Object -Property 'Name' |
            Set-AttributeValue
```